src/entities/nestedExit/getNestedExitCalls.ts

```diff
@@ -0,0 +1,193 @@
+import { Token } from '../token';
+import { NestedExitInput, NestedExitCall } from './types';
+import { NestedPool, NestedPoolState } from '../types';
+import { TokenAmount } from '../tokenAmount';
+import { Address } from '../../types';
+
+export const getNestedExitCalls = (
+    {
+        bptAmountIn,
+        chainId,
+        accountAddress,
+        tokenOut,
+        useNativeAssetAsWrappedAmountOut = false,
+        toInternalBalance = false,
+    }: NestedExitInput,
+    { pools }: NestedPoolState,
+): { bptAmountIn: TokenAmount; calls: NestedExitCall[] } => {
+    const isProportional = tokenOut === undefined;
```

Comment:
This is common for `getNestedExitCalls` and `parseNestedExitCall`, we could do it early and explicitly in NestedExit.query and pass a boolean or enum as a param to those functions which would make it clearer.